auraed/src/ebpf/tracepoint_programs/mod.rs

```diff
@@ -0,0 +1,44 @@
+/* -------------------------------------------------------------------------- *\
+ *        Apache 2.0 License Copyright © 2022-2023 The Aurae Authors          *
+ *                                                                            *
+ *                +--------------------------------------------+              *
+ *                |   █████╗ ██╗   ██╗██████╗  █████╗ ███████╗ |              *
+ *                |  ██╔══██╗██║   ██║██╔══██╗██╔══██╗██╔════╝ |              *
+ *                |  ███████║██║   ██║██████╔╝███████║█████╗   |              *
+ *                |  ██╔══██║██║   ██║██╔══██╗██╔══██║██╔══╝   |              *
+ *                |  ██║  ██║╚██████╔╝██║  ██║██║  ██║███████╗ |              *
+ *                |  ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝ |              *
+ *                +--------------------------------------------+              *
+ *                                                                            *
+ *                         Distributed Systems Runtime                        *
+ *                                                                            *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ *   Licensed under the Apache License, Version 2.0 (the "License");          *
+ *   you may not use this file except in compliance with the License.         *
+ *   You may obtain a copy of the License at                                  *
+ *                                                                            *
+ *       http://www.apache.org/licenses/LICENSE-2.0                           *
+ *                                                                            *
+ *   Unless required by applicable law or agreed to in writing, software      *
+ *   distributed under the License is distributed on an "AS IS" BASIS,        *
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ *   See the License for the specific language governing permissions and      *
+ *   limitations under the License.                                           *
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+
+use aurae_ebpf_shared::Signal;
+pub use perf_event_listener::PerfEventListener;
+pub use tracepoint_program::TracepointProgram;
+
+mod perf_event_listener;
+mod tracepoint_program;
+
+pub struct SignalSignalGenerateTracepointProgram;
```

Comment:
I thought of naming this `PosixSignalTracepointProgram`, but I assume there was a reason for the naming we had?

Reply:
The event we are tracing is `signal:signal_generate` where `signal` is the subsystem and `signal_generate` is the actual event. This name makes this mapping to the Linux trace event obvious, but I agree it's not very easy on the eye. Once we have a bunch more of these tracepoints we could put them in a module per subsystem so we can get rid of the subsystem prefix.